src/dependency_injector/providers.pyx

```diff
@@ -4645,4 +4734,4 @@ cpdef str _class_qualname(object instance):
     name = getattr(instance.__class__, '__qualname__', None)
     if not name:
         name = '.'.join((instance.__class__.__module__, instance.__class__.__name__))
-    return name
\ No newline at end of file
+    return name
```

Comment:
👍

Reply:
Vim sometimes strips new lines at the end of files.